epix/clustering.py

```diff
@@ -25,39 +26,75 @@ def find_cluster(interactions, cluster_size_space, cluster_size_time):
         df.append(ak.to_pandas(interactions[key], anonymous=key))
     df = pd.concat(df, axis=1)
 
-    # Splitting into individual events and cluster:
+    # Splitting into individual events and apply time clustering:
     groups = df.groupby('entry')
-    groups = groups.apply(lambda x: _find_cluster(x, cluster_size_space, cluster_size_time))
+    df["time_cluster"] = np.concatenate(groups.apply(lambda x: simple_1d_clustering(x.t.values, cluster_size_time)))
+
+    # Splitting into individual events and time cluster and apply space clustering space:
+    groups = df.groupby(['entry', 'time_cluster'])
+    groups = groups.apply(lambda x: _find_cluster(x, cluster_size_space))
+
+    for i in np.unique(groups.index.get_level_values(0)):
```

Comment:
Is not this the same as np.arange(len(groups)) ?

Reply:
No it is sightly different. We are dealing here with the highest level index of this multi-indexed dataframe. So we have `n` times `0` and then `m` times `1` and so on. This can look something like: `array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 3, 3, 3, 3])`. I just need to loop here over `[0, 1, 2, 3]` and can achieve this here with the `np.unique` function.

Reply:
Is still do not get it, why is this different from e.g. `np.arange(4)?

Reply:
This is in principle the same. One could also use something like: `np.arange(groups.index.get_level_values(0).max())`. This should be equal to `np.arange(4)` in this example here.